src/main/java/org/folio/circulation/domain/RequestQueue.java

```diff
@@ -51,24 +56,18 @@ public boolean isRequestedByOtherPatron(User requestingUser) {
   public void add(Request newRequest) {
     requests = new ArrayList<>(requests);
     requests.add(newRequest);
-    orderRequests();
+    reSequenceRequests();
   }
 
   public void remove(Request request) {
     requests = requests.stream()
       .filter(r -> !r.getId().equals(request.getId()))
       .collect(Collectors.toList());
     request.removePosition();
-    orderRequests();
+    reSequenceRequests();
   }
 
-  private void orderRequests() {
-    requests = requests.stream()
-      // order by date descending
-      .sorted((req1, req2) -> req1.getRequestDate().compareTo(req2.getRequestDate()))
-      // order by (request status = "Open - In transit" or "Open - Awaiting pickup") first
-      .sorted((req1, req2) -> Boolean.compare(req2.isNotDisplaceable(), req1.isNotDisplaceable()))
-      .collect(Collectors.toList());
+  private void reSequenceRequests() {
     final AtomicInteger position = new AtomicInteger(1);
```

Comment:
Why does this need to use an atomic integer?

Reply:
The outer variable must be final, in order to use it in lambda. We can not just use `position++` or similar and have to use AtomicInteger because it is mutable and the reference can be final.